File: strategy_pattern/ruby/behaviors/quack_behavior.rb
module QuackBehavior
  class Quack
    def quack
      puts 'Quack! Quack!'
    end
  end

  class Squeak
    def quack
      puts 'Squeak!'
    end
  end

  class MuteQuack
    def quack
      puts '<< Silence >>'
    end
  end
end

File: strategy_pattern/go/behavior/quack_behavior.go
package behavior

// Strategy Interface
type QuackBehavior interface {
	Quack()
}

// Concrete Strategies
type Quack struct{}
type Squeak struct{}
type MuteQuack struct{}

func (q *Quack) Quack() {
	println("Quack")
}

func (q *Squeak) Quack() {
	println("Squeak")
}

func (q *MuteQuack) Quack() {
	println("<< Silence >>")
}
